webhook/resources/volumeattachment/validator.go

```diff
@@ -0,0 +1,87 @@
+package volumeattachment
+
+import (
+	"fmt"
+	"github.com/longhorn/longhorn-manager/datastore"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/types"
+	"github.com/longhorn/longhorn-manager/webhook/admission"
+	werror "github.com/longhorn/longhorn-manager/webhook/error"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"reflect"
+)
+
+type volumeAttachmentValidator struct {
+	admission.DefaultValidator
+	ds *datastore.DataStore
+}
+
+func NewValidator(ds *datastore.DataStore) admission.Validator {
+	return &volumeAttachmentValidator{ds: ds}
+}
+
+func (v *volumeAttachmentValidator) Resource() admission.Resource {
+	return admission.Resource{
+		Name:       "volumeattachments",
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   longhorn.SchemeGroupVersion.Group,
+		APIVersion: longhorn.SchemeGroupVersion.Version,
+		ObjectType: &longhorn.VolumeAttachment{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+			admissionregv1.Update,
+		},
+	}
+}
+
+func (v *volumeAttachmentValidator) Create(request *admission.Request, newObj runtime.Object) error {
+	va, ok := newObj.(*longhorn.VolumeAttachment)
+	if !ok {
+		return werror.NewInvalidError(fmt.Sprintf("%v is not a *longhorn.VolumeAttachment", newObj), "")
+	}
+
+	if err := verifyAttachmentTicketIDConsistency(va.Spec.AttachmentTickets); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (v *volumeAttachmentValidator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) error {
+	oldVA, ok := oldObj.(*longhorn.VolumeAttachment)
+	if !ok {
+		return werror.NewInvalidError(fmt.Sprintf("%v is not a *longhorn.VolumeAttachment", oldObj), "")
+	}
```

Comment:
Why this will happen? this should be called by Kubernetes directly and the type is assumed. Probably we don't need this as other validators. 

The same for new obj.

Reply:
I was thinking that our webhook should be more defensive and not crash even if kubeAPI sends us a wrong object. But yeah, that probably never happens. 

Fixed, Thank you